src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -64,6 +64,80 @@
     <EditorConfigFiles Include="@(PotentialEditorConfigFiles->Exists())" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
   </ItemGroup>
 
+  <!--
+    ========================
+    Property/metadata global .editorconfig Support
+    ========================
+    
+    Generates a global analyzer config that contains the evaluation of requested MSBuild properties and item metadata
+    
+    Requested properties/items are requested via item groups like:
+   
+      <AnalyzerProperty Include="PropertyNameToEval" />
+      <AnalyzerItemMetadata Include="ItemType" MetadataName="MetaDataToRetrieve" />
+    -->
+  <UsingTask TaskName="Microsoft.CodeAnalysis.BuildTasks.GenerateMSBuildAnalyzerConfig" AssemblyFile="$(MSBuildThisFileDirectory)Microsoft.Build.Tasks.CodeAnalysis.dll" />
+
+  <PropertyGroup>
+    <GeneratedMSBuildAnalyzerConfigFile Condition="'$(GeneratedMSBuildAnalyzerConfigFile)' ==''">$(IntermediateOutputPath)$(MSBuildProjectName).GeneratedMSBuildAnalyzerConfig.editorconfig</GeneratedMSBuildAnalyzerConfigFile>
+    <GenerateMSBuildAnalyzerConfigFile Condition="'$(GenerateMSBuildAnalyzerConfigFile)' == ''">true</GenerateMSBuildAnalyzerConfigFile>
+  </PropertyGroup>
+
+  <Target Name="GenerateMSBuildAnalyzerConfigFileShouldRun"
+          BeforeTargets="GenerateMSBuildAnalyzerConfigFile"
+          Condition="'$(Language)'=='VB' or '$(Language)'=='C#'">
+    <PropertyGroup>
+      <_GeneratedAnalyzerConfigHasItems Condition="'@(AnalyzerItemMetadata->Count())' != '0'">true</_GeneratedAnalyzerConfigHasItems>
+      <_GeneratedAnalyzerConfigShouldRun Condition="'$(GenerateMSBuildAnalyzerConfigFile)' == 'true' and ('$(_GeneratedAnalyzerConfigHasItems)' == 'true' or '@(AnalyzerProperty->Count())' != '0')">true</_GeneratedAnalyzerConfigShouldRun>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="GenerateMSBuildAnalyzerConfigFile"
```

Comment:
>GenerateMSBuildAnalyzerConfigFile [](start = 16, length = 33)

The name of this target seems confusing, since it's also used as a property above. Or is this dual meaning how things are typically done in MSBuild?

Reply:
Yeah, its weird, but there is precedent in other targets to have a target/property named the same thing. 

---
In reply to: [419682753](https://github.com/dotnet/roslyn/pull/43617#discussion_r419682753) [](ancestors = 419682753)